public/app/features/explore/Explore.tsx

```diff
@@ -315,14 +316,14 @@ function mapStateToProps(state: StoreState, { exploreId }: ExploreProps) {
     urlState,
     update,
     queryErrors,
-    mode,
     isLive,
   } = item;
 
-  const { datasource, queries, range: urlRange, ui } = (urlState || {}) as ExploreUrlState;
+  const { datasource, queries, range: urlRange, mode: urlMode, ui } = (urlState || {}) as ExploreUrlState;
   const initialDatasource = datasource || store.get(LAST_USED_DATASOURCE_KEY);
   const initialQueries: DataQuery[] = ensureQueries(queries);
   const initialRange = urlRange ? getTimeRangeFromUrl(urlRange, timeZone).raw : DEFAULT_RANGE;
+  const mode = urlMode || ExploreMode.Metrics;
```

Comment:
This should use `mode` from props instead of `Metrics` if `urlMode` is undefined. `mode` from props should be the selected mode from loading the datasource. Thinking a bit more, `supportedModes` from props contains the `modes` support for the selected datasource so that's need to be validated somewhere, similar to how the reduces handles the `updateDatasourceInstanceAction` action.